lib/rules/require-prop-types.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * @fileoverview Prop definitions should be detailed
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function create (context) {
+  // variables should be defined here
+
+  // ----------------------------------------------------------------------
+  // Helpers
+  // ----------------------------------------------------------------------
+
+  function getPropTypes (componentProperties) {
+    const node = componentProperties
+      .filter(p =>
+        p.type === 'Property' &&
+        p.key.type === 'Identifier' &&
+        p.key.name === 'props'
+      )[0]
+
+    if (!node) {
+      return {
+        type: null,
+        nodes: []
+      }
+    }
+
+    let nodes = []
+    const type = node.value.type
+    if (type === 'ObjectExpression') { // props: {
+      nodes = node.value.properties.map(cp => {
+        if (cp.type !== 'Property') {
+          return { key: null, node: cp, hasType: true }
+        }
+        const key = cp.key.name
+        let hasType = true
+        if (cp.value.type === 'ObjectExpression') { // foo: {
+          hasType = !!(cp.value.properties
+            .filter(p =>
+              p.key.type === 'Identifier' &&
+              p.key.name === 'type' &&
+              (
+                p.value.type !== 'ArrayExpression' ||
+                p.value.elements.length > 0
+              )
+            )[0])
+        } else if (cp.value.type === 'ArrayExpression') { // foo: [
+          hasType = cp.value.elements.length > 0
+        } else if (cp.value.type === 'FunctionExpression' || cp.value.type === 'ArrowFunctionExpression') {
+          hasType = false
+        }
+        return { key, node: cp, hasType }
+      })
+    } else if (type === 'ArrayExpression') { // props: [
+      // nodes = node.elements
+    }
+
+    return {
+      type,
+      node,
+      nodes
+    }
+  }
+
+  // ----------------------------------------------------------------------
+  // Public
+  // ----------------------------------------------------------------------
+
+  return utils.executeOnVueComponent(context, (obj) => {
+    const data = getPropTypes(obj.properties)
+    if (data.type === 'ObjectExpression') {
+      data.nodes.forEach(cp => {
+        if (!cp.hasType) {
+          context.report({
+            node: cp.node || data.node,
+            message: 'Prop "{{name}}" definitions should always be as detailed with at least type(s).',
```

Comment:
What about this message instead?
`Prop "{{name}}" should define at least it's type.`